src/System.IO.Compression/src/System/IO/Compression/DeflateStream.Unix.cs

```diff
@@ -1,13 +1,27 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.IO.Compression
 {
     public partial class DeflateStream : Stream
     {
         private static WorkerType GetDeflaterType()
         {
-            return WorkerType.Managed; // TODO: Switch to ZLib once interop is worked out
+            try
+            {
+                // Make any P/Invoke into zlib to ensure we're able to find and use it.
+                // If we are, then use zlib.
+                Interop.zlib.zlibCompileFlags(); 
+                return WorkerType.ZLib;
+            }
+            catch
+            {
+                // Otherwise, fallback to managed implementation if zlib isn't available
+                Debug.Fail("zlib unavailable");
```

Comment:
I was actually thinking InternalsVisibleTo ;)  But, sure, we could do a COMPlus_ environment variable or something if that's more appropriate.  Whatever we did, I think we'd want to make it dynamic so that a given test could force a fallback.  I believe the System.Reflection.Metadata tests use InternalsVisibleTo with a flag to force / test certain fallback cases.